### ISSUE TITLE
Basis Plugin Dateien

### ISSUE THREAD

**Issue by Cyperghost** [DevTekDE/WSC-API#1]
Erstellung der Basis Dateien für das WSC3.0 Plugin.

**Comment by Cyperghost** [comment]
Müsste nur jemand als Feature markieren sowie eine passende Branch erstellen, dazu habe ich keine rechte...

**Comment by ZargorNET** [comment]
Kannst du kein Pull-Request machen?

**Comment by Cyperghost** [comment]
Doch, aber immer nur in die dev branch. Oder ist dies so gewollt.

**Comment by ZargorNET** [comment]
Hab dir 'nen neuen Branch erstellt

**Comment by DevTek314** [comment]
Wollen wir das Plugin wirklich nur auf Discord beschränken oder eine komplette REST-API für alles zur Verfügung stellen?
Dann hätte das Forum endlich mal eine richtig gute API.

**Comment by Cyperghost** [comment]
Welche Möglichkeiten möchtest du über die API den anbieten?
Möglich ist alles.

**Comment by DevTek314** [comment]
Ja vorgestellt hatte ich mir eine REST-API per HTTP-Requests.
Das man nicht direkt zu Anfang alle Funktionen, welche das Forum besitzt implementieren kann ist klar, aber wir könnten das Projekt ja immer weiter führen und die API immer um Funktionen erweitern.

**Comment by ZargorNET** [comment]
Müssten dann aber von diesem Projekt noch einen Fork machen(sobald das Projekt hier weit genug fortgeschritten ist), da der Bot muss ja auch requests vom Plugin erhalten

**Comment by FelixKlauke** [comment]
Jersey mit HTTP Container wäre meine Wahl. Etwas ähnliches haben wir bereits im Discordbot repo diskutiert.

**Comment by Nathalie0hneHerz** [comment]
Jersey mit HTTP Container wäre meine Wahl. Etwas ähnliches haben wir bereits im Discordbot repo diskutiert.

**Comment by Cyperghost** [comment]
@DevTek314 
Dann ist es ja eigentlich besser, wenn wir die von MysterCode nehmen. Die hat soweit ich sehe eigentlich alles nötige drinnen.
https://github.com/MysteryCode/de.mysterycode.wcf.rest

**Comment by DevTek314** [comment]
ok, die kannte ich noch nicht.
Aber die wäre ja auch nicht auf Jersey angepasst, oder?

**Comment by Cyperghost** [comment]
Das ist eine normale API, die kann mit jedem normale WEB API Tool angesteuert werden.
Ansonsten ist es natürlich kein Problem eine eigene zu erstellen.

**Comment by DevTek314** [comment]
Ich denke eine eigene wäre schon etwas, da man diese dann auch besser an den Discord Bot anpassen kann.

**Comment by Cyperghost** [comment]
Okay. Dann müsste nur geklärt werden. Soll diese lediglich für den Discord Server sein oder auch mehr machen? Falls mehr, was genau. Da man hier dann einen Namen anpassen sollte, aber das sollte ja auch im nach hin möglich sein.
